api/envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.proto

```diff
@@ -40,6 +40,7 @@ message HttpConnectionManager {
     // (Envoy does not require HTTP/2 to take place over TLS or to use ALPN.
     // Prior knowledge is allowed).
     HTTP2 = 2;
+    HTTP3 = 3;
```

Comment:
comment

Reply:
Along with the comment, can we also not-implemented-hide this until we believe we have a working MVP implementation? Also, how does this interact with AUTO? If AUTO is configured will it work for QUIC? I'm wondering what the valid QUIC configs will be as I think the config needs to be fairly specific in terms of how it will be specified in the MVP?

Reply:
> Along with the comment, can we also not-implemented-hide this until we believe we have a working MVP implementation? 

done

> Also, how does this interact with AUTO? If AUTO is configured will it work for QUIC? 

The config has to specify HTTP3 as codec_type. AUTO will not work for QUIC as it inspects the first a few bytes of the `data` argument passed to onData(). And QUIC just doesn't execute this code pass.

> I'm wondering what the valid QUIC configs will be as I think the config needs to be fairly specific in terms of how it will be specified in the MVP?

It's already defined in api/envoy/api/v2/listener/quic_config.proto. And it will be specified in Listener.udp_listener_config. A sample config is here: https://github.com/envoyproxy/envoy/pull/8496/files#diff-6e3a243807199766bdae36c0c2f242ebR116